ql/models/shortrate/calibrationhelpers/swaptionhelper.cpp

```diff
@@ -141,20 +152,24 @@ namespace QuantLib {
     void SwaptionHelper::performCalculations() const {
 
         Calendar calendar = index_->fixingCalendar();
+        Natural effSettlDays =
+            settlementDays_ == Null<Size>() ? index_->fixingDays() : settlementDays_;
 
         Date exerciseDate = exerciseDate_;
         if (exerciseDate == Null<Date>())
             exerciseDate = calendar.advance(termStructure_->referenceDate(),
                                             maturity_,
                                             index_->businessDayConvention());
 
-        Date startDate = index_->valueDate(index_->fixingCalendar().adjust(exerciseDate));
+        Date startDate = calendar.advance(exerciseDate,
+                                          effSettlDays, Days,
+                                          index_->businessDayConvention());
```

Comment:
Hmm, indexes such as LIBORs might have particular rules for `valueDate`.  I would keep the previous call to `valueDate` if `settlementDays_ == Null<Size>()` and only switch to this `calendar.advance` otherwise.

Reply:
makes sense, updated.